USS Mayflower

VividPups opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
Adds a New Expedition/Bounty hunter ship.  The USS Mayflower 

## Why / Balance
As far as I was able to see the only Bounty hunter/Expedition ship was the KC Sprinter. So I would like to add more that is more dedicated in bounty hunting/ Expeditions

## Technical details
<!-- If this is a code change, summarize at a high level how your new code works. This makes it easier to review. -->

## Media
![image](https://github.com/new-frontiers-14/frontier-station-14/assets/150621839/ae7e1dc7-4e47-44c7-b10c-782b06397195)


- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase


**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- add: Added The USS Mayflower

dvir001 commented:
![image](https://github.com/new-frontiers-14/frontier-station-14/assets/39403717/bbb3c73f-dcfb-45ce-8d1c-871c0943ffc5)
Names need to match, lowercase only

dvir001 commented on Resources/Prototypes/_NF/Shipyard/mayflower.yml:
```diff
@@ -0,0 +1,28 @@
+      stationProto: StandardFrontierExpeditionVessel
+      components:
+        - type: StationNameSetup
+          mapNameTemplate: 'USS Mayflower{1}'
```
Name format missing space

VividPups replied:
what do you mean by that? Where is the space post to be at?

Cheackraze replied:
> what do you mean by that? Where is the space post to be at?

between Mayflower and {1}

VividPups replied:
ooo ok also... i just want to say I appearnlty mess this up really big so I am right now fixing all of this with the help of minty

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - pos: -3.5,-5.5
+      parent: 1
+      type: Transform
+- proto: LockerCaptain
```
Please keep it to base line lockers, pre filled.

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - canCollide: False
+      type: Physics
+    - type: InsideEntityStorage
+- proto: ClothingOuterCoatHoSTrench
```
Hos gear?? this is sec gear.

VividPups replied:
you can buy the sheriffs hat from an BountyVending and the trenchcoat is just renamed like it is on the sprinter

dvir001 replied:
Its getting fixed, the hat will need to get a new version with a new name..
Ill look into adding the 2 as new versions to be used for it, might as well.

Cheackraze replied:
i believe its in now with the merc gear rework

VividPups replied:
I have change the  custom bounty hunter name hos cloths To actual  Bounty hunter cloths

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+          component: Armor
+        title: null
+      type: GroupExamine
+- proto: ClothingOuterStraightjacket
```
Not sure if we want this mapped ever.

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - pos: -5.5,-1.5
+      parent: 1
+      type: Transform
+- proto: ClothingUniformJumpskirtDetective
```
Detective gear?

VividPups replied:
the hard worn suit is an jumpsuit that you can buy from the repdrobe that all civils have access too and its on the sprinter ship another Expedition ship

VividPups replied:
these has been corrected

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - canCollide: False
+      type: Physics
+    - type: InsideEntityStorage
+- proto: ClothingUniformJumpsuitDetective
```
Detective gear?

VividPups replied:
the hard worn suit you can buy from the Repdrobe and its on the sprinter ship

VividPups replied:
this has been change to bounty

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - ShutdownSubscribers:
+      - 460
+      type: DeviceNetwork
+- proto: FloorDrain
```
Drain in a merc and medical ship?

VividPups replied:
well the sprinter has one including the dart, anchor even tho the dart and anchor isnt an merc ship

VividPups replied:
Drain has been removed

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - pos: 0.5,-5.5
+      parent: 1
+      type: Transform
+- proto: LockerSalvageSpecialistFilledHardsuit
```
Salvage locker also?

VividPups replied:
all expedition ships that i have send has an salvage locker but the anchor and the dart that i have seen?

Cheackraze replied:
this is fine dw

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+      pos: -0.5,-6.5
+      parent: 1
+      type: Transform
+- proto: WeaponRevolverMateba
```
Detective gun?

VividPups replied:
also on the sprinter

dvir001 replied:
Getting redone with rework.

VividPups replied:
and the gourd as an Krammar (sec gun) and the anchor has MK38s (sec guns)

dvir001 replied:
Ill add it to the list, we will probably end up adding new safes to just allow ships to pick from to keep some baseline

dvir001 commented on Resources/Maps/Shuttles/Mushroom.yml:
```diff
@@ -0,0 +1,5827 @@
+    - pos: -5.4586143,-1.5184155
+      parent: 1
+      type: Transform
+- proto: WeaponShotgunSawn
```
Pre cut shotgun?

VividPups replied:
this is on the sprinter

dvir001 replied:
Getting redone with rework.

VividPups replied:
and i believe like 2 other ships

dvir001 reviewed (commented):
This ship has alot of security gear and gamery gear as of now.

Cheackraze commented:
we are indeed in the middle of reworking how weapons and such are given through ships. We are about to have 'gun lockers' in 3 tiers, corresponding to the ship's value, so it would be best not to map individual guns right now in favor of those cabinets in order to keep it in line with the rest. To be fair, however, we did determine that Kammerers are more of an advanced-civilian weapon in their current state, with the Security grade shotgun becoming the enforcer.

Also, the 'bounty hunter' variants of the items found on the sprinter, are now official spawned entities you can use again.  I super appreciate your patience here, but We've been working to add the things to make sure this ship can be everything its designed to be! Looking forward to a possible update

VividPups commented:
@Cheackraze  I do apparate the Responds with the extra info about what is going on behind the scene's and what is to come, Im sure you can tell by now; The USS Mayflower was Heavily influence by the Sprinter and I wanted to make more Bounty hunter style ships. Now at this point I kinda of step away from SS13 as a whole due to some stuff that has happen on SS14 I made it clear to some people there is a good chance I wont return. However, I would still like to continue to atleast finish this ship and it seems like the best Course of action is to wait for some of thoses things to be release

Cheackraze commented:
> @Cheackraze I do appreciate the responses with the extra info about what is going on behind the scenes and what is to come, I am sure you can tell by now that the USS Mayflower was Heavily influenced by the Sprinter and I wanted to make more Bounty Hunter style ships. Now at this point I kinda of stepped away from SS13 as a whole due to some stuff that has happened on SS14 I made it clear to some people that there is a good chance I won't return. However, I would still like to continue to at least finish this ship and it seems like the best Course of action is to wait for some of those things to be released

all of the mercenary gear has been updated, feel free to update the shuttle with the new items so I can merge it. Let me know or ask on discord if you need help merging the latest updates into your working branch, so that you get access to the new items.

Cheackraze commented:
so, full review on whats presented. I have 2 issues with it overall, but they are mostly minor. One is the lack of dirt. Since the theme here is abandoned/cobweb infested/broken down, I feel that adding (cleanable) dirt decals throughout the ship will really bring it all together. And second, is the use of kitty corner wall design. 

![mayflower](https://github.com/new-frontiers-14/frontier-station-14/assets/71046427/09a1b9c6-3c6b-4227-8bbc-8dde94fb4a95)

Although its tempting to try to save space using this, it has a poor visual effect on the walls and lighting system in the game. With icon smoothing on walls, you get strange looking walls and can 'see' on the other side of the walls where it doesnt connect and looks bad. Another issue is just the fact that it creates weak points in the design, and have the potential to react funky with projectiles passing 'through' that point in the wall. Among other things.  I feel that you definitely have plenty of space inside the ship to adjust some of the rooms and shapes to avoid this issue, while still keeping true to your design goals 100%. Otherwise, I like the design and idea behind it, and this will likely join the bison and SV series shuttles (maybe a few others like kestrel, maybe pathfinder...) in a 'scrapyard' shipyard catalogue.

VividPups commented:
So pretty much what happened was. I appeared wasn't doing it right in a way where I couldn't update the local server with the new release stuff. Which in respond. I had to create a new Ship file for the game shipyard (which now when I buy from the shipyard it just freezes the game completely).  However.  I changed the Kitty Corners mark on the picture and added dirt to all the tiles. so it look more broken down and old. At this point I have no idea what is causing the crashing now. I had ask the discord about it and right now as of now they don't know why it is crashing

VividPups commented:
Ok I fix it and as I try to figure out how to update all of this Have an Updated look of the ship 
![Mayflower](https://github.com/new-frontiers-14/frontier-station-14/assets/150621839/fd0271f1-a5dc-44fa-8201-134f0976044a)

VividPups commented:
Ok, I think I did it but not sure?

VividPups commented:
I did Press Push multiple times and I guess it didn't work...I'll try again after work and if there isn't any change I'll make a new. I believe with how much the change happen last night with it could be the problem.

VividPups commented:
So with the Push failing so much. I am just going to close this one and then make a new one with the updated everything since for some reason this one isn't updating

VividPups commented:
So I am getting an Push failed error (trying to make the new one)
... Unable to Push to the remote repositiory because your local branch is behind the remote branch. Update your branch by pulling before pushing.

but both of the pulling is gray out and im not sure how to fix that

VividPups commented:
At this point im at a lost.. and don't know what to do to be able to make the new Pull request since this one isn't allowing me to update it

VividPups commented:
https://github.com/new-frontiers-14/frontier-station-14/pull/734#issue-2053183009

Link to the new Request with the updates that was ask/told to do.
